modules/cudf/test/cudf-column-tests.ts

```diff
@@ -143,3 +149,63 @@ test('Column.nansToNulls', () => {
   const expected = [1, 3, null, 4, 2, 0];
   expect([...Series.new(result)]).toEqual(expected);
 });
+
+describe('Column.setNullMask', () => {
+  const arange = (length: number) => Array.from({length}, (_, i) => i);
+  const makeTestColumn            = (length: number) =>
+    new Column({type: new Int32, length, data: arange(length)});
+
+  const validateSetNullMask = (col: Column, length: number, expectedNullCount: number, newNullMask: any, ...args: [any?]) => {
+    expect(col.length).toBe(length);
+    expect(col.nullCount).toBe(0);
+
+    col.setNullMask(newNullMask, ...args);
+
+    expect(col.length).toBe(length);
+    expect(col.nullCount).toBe(expectedNullCount);
+    expect(col.hasNulls).toBe(expectedNullCount > 0);
+    expect(col.nullable).toBe(newNullMask != null);
+    if (newNullMask == null) {
+      expect(col.mask.byteLength).toBe(0);
+    } else {
+      expect(col.mask.byteLength).toBe((((length >> 3) + 63) & ~63) || 64);
+    }
+  };
+
+  test('recomputes nullCount (all null)',
+       () => { validateSetNullMask(makeTestColumn(4), 4, 4, new DeviceBuffer(64)); });
+
+  test('recomputes nullCount (all valid)',
+       () => { validateSetNullMask(makeTestColumn(4), 4, 0, new Uint8Buffer(8).fill(255)); });
+
+  test('uses the new nullCount',
+       () => { validateSetNullMask(makeTestColumn(4), 4, 4, new DeviceBuffer(8), 4); });
+
+  test('clamps the new nullCount to length',
+       () => { validateSetNullMask(makeTestColumn(4), 4, 4, new DeviceBuffer(8), 8); });
+
+  test('resizes when mask is smaller than 64 bytes',
+       () => { validateSetNullMask(makeTestColumn(4), 4, 4, new DeviceBuffer(4)); });
+
+  test('Passing null resets to all valid',
+       () => { validateSetNullMask(makeTestColumn(4), 4, 0, null); });
+
+  test('accepts Arrays of numbers', () => {
+    validateSetNullMask(makeTestColumn(4), 4, 4, [0, 0, 0, 0]);
+  });
+
+  test('accepts Arrays of bigints', () => {
+    validateSetNullMask(makeTestColumn(4), 4, 4, [0n, 0n, 0n, 0n]);
+  });
+
+  test('accepts Arrays of booleans', () => {
+    validateSetNullMask(makeTestColumn(4), 4, 4, [false, false, false, false]);
+  });
```

Comment:
Maybe good to explicitly test a not-all-null case and the no-nulls case with array inputs?